components/fxa-client/src/internal/mod.rs

```diff
@@ -54,6 +54,11 @@ pub struct FirefoxAccount {
     devices_cache: Option<CachedResponse<Vec<http_client::GetDeviceResponse>>>,
     auth_circuit_breaker: AuthCircuitBreaker,
     telemetry: FxaTelemetry,
+    /// Application-supplied event listener.  Wrapped with an Arc since it's shared with StateManager.
+    ///
+    /// Arc<Box<dyn T>> is kind of silly, but I couldn't figure out how to properly move the inner
+    /// value out of a Box and into an Arc.
+    pub event_listener: Option<Arc<Box<dyn EventListener>>>,
```

Comment:
related to my earlier comment: what if it's only owned by this layer and not shared with the StateManager? That might add some complexity in passing data but could simplify that event handling